File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        int n = s.length();
        int[] hm = new int[26];
        char[] charMp = s.toCharArray();

        for(int i = 0; i < n; i++){
            hm[charMp[i]-'a']++;
        }

        for(int i = 0; i < n; i++){
            if(hm[charMp[i]-'a'] == 1){
                return i;
            }
        }
        return -1;

    }
}

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.go
func firstUniqChar(s string) int {
    var hm [26]int;

    for i:= range s {
        hm[s[i]-'a']++;
    }

    for  i:= range s {
        if hm[s[i]-'a'] == 1{
            return i;
        }
    }
    return -1;
}